pkg/catalogsourceconfig/catalogsourcebuilder.go

```diff
@@ -28,13 +28,29 @@ func (b *CatalogSourceBuilder) WithTypeMeta() *CatalogSourceBuilder {
 // WithMeta sets basic TypeMeta and ObjectMeta.
 func (b *CatalogSourceBuilder) WithMeta(name, namespace string) *CatalogSourceBuilder {
 	b.WithTypeMeta()
-	b.cs.ObjectMeta = metav1.ObjectMeta{
-		Name:      name,
-		Namespace: namespace,
+	objectMeta := b.cs.GetObjectMeta()
+	if objectMeta == nil {
+		b.cs.ObjectMeta = metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		}
+	} else {
+		// This is to account for the scenario where WithVisibilityMeta() was called before
+		b.cs.SetName(name)
+		b.cs.SetNamespace(namespace)
 	}
 	return b
 }
 
+// WithVisibilityMeta sets basic TypeMeta and ObjectMeta with the "olm-visibility" label.
+func (b *CatalogSourceBuilder) WithVisibilityMeta(name, namespace string) *CatalogSourceBuilder {
+	b.WithTypeMeta()
+	b.WithMeta(name, namespace)
+	// Hint for OLM Packages UI to hide this CatalogSource
+	b.cs.ObjectMeta.Labels = map[string]string{"olm-visibility": "hidden"}
```

Comment:
Is this separate from the Datastore label? I thought they were the same until I saw the label string was different.

Reply:
Yes this is different from the datastore label.